python/lsst/daf/butler/formatters/astropyTable.py

```diff
@@ -0,0 +1,92 @@
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+__all__ = ("AstropyTableFormatter", )
+
+from typing import (
+    Any,
+    Optional,
+    Type,
+)
+import os.path
+
+from .file import FileFormatter
+
+
+class AstropyTableFormatter(FileFormatter):
+    """Interface for reading and writing astropy.Table objects
+    in either ECSV or FITS format
+    """
+    supportedWriteParameters = frozenset({"format"})
+    supportedExtensions = frozenset({".ecsv", ".fits", ".fit"})
+
+    @property
+    def extension(self) -> str:  # type: ignore
+        # Typing is ignored above since this is a property and the
+        # parent class has a class attribute
+
+        # Default to ECSV but allow configuration via write parameter
+        format = self.writeParameters.get("format", "ecsv")
+        if format == "ecsv":
+            return ".ecsv"
+        elif format == "fits":
+            return ".fits"
+        raise RuntimeError(f"Requested file format '{format}' is not supported for Table")
+
+    def _readFile(self, path: str, pytype: Optional[Type[Any]] = None) -> Any:
+        """Read a file from the path in a supported format format.
+
+        Parameters
+        ----------
+        path : `str`
+            Path to use to open the file.
+        pytype : `type`
+            Class to use to read the serialized file.
+
+        Returns
+        -------
+        data : `object`
+            Instance of class ``pytype`` read from serialized file. None
+            if the file could not be opened.
+        """
+        if not os.path.exists(path) or pytype is None:
+            return None
+
+        return pytype.read(path)
+
+    def _writeFile(self, inMemoryDataset: Any) -> None:
+        """Write the in memory dataset to file on disk.
+
+        Parameters
+        ----------
+        inMemoryDataset : `object`
+            Object to serialize.
+
+        Raises
+        ------
+        Exception
+            The file could not be written.
+        """
+        format_str = self.writeParameters.get('format', None)
+        if format_str:
+            inMemoryDataset.write(self.fileDescriptor.location.path, format=format_str)
+        else:
+            inMemoryDataset.write(self.fileDescriptor.location.path)
```

Comment:
is this else branch needed? I'm not entirely sure how `write()` works and the help docs don't really tell me, but if `format=None` is used is that the same as not specifying format? What does absence of format mean? Does it use the file extension? If it uses the file extension why do we need both options here?

Reply:
Good questions.  I guess this came from my wanting to be safe and I think this will work no matter how things work internal to `Table`

Reply:
I'd prefer to just leave it unless you strongly object

Reply:
I don't understand what the else is doing here. In the extension method we guarantee we will have a valid supported extension.  Then in the write method there is the same `.get()` but here the default is `None`. Why isn't the default the same as for `extension` above?  What does not specifying `format` actually mean? If there was a package variable with `DEFAULT_FORMAT = "ecsv"` and you used that in both extension and the `.get` here, wouldn't that remove the else completely?

Reply:
You're absolutely right.  I don't think we need the `format` option at all.  `astropy.Table` discovers the correct output format from the extension as we would expect.  I'm going to remove the else.